contrib/win32/win32compat/win32_pty.c

```diff
@@ -154,34 +154,38 @@ int exec_command_with_pty(int * pid, char* cmd, int in, int out, int err, unsign
 	si.hStdError = (HANDLE)w32_fd_to_handle(err);
 	si.lpDesktop = NULL;
 
-	if (is_conpty_supported())
-		return CreateConPty(cmd_w, (short)si.dwXCountChars, (short)si.dwYCountChars, si.hStdInput, si.hStdOutput, ttyh, &pi);
-
-	/* launch via  "ssh-shellhost" -p command*/
-
-	_snwprintf_s(pty_cmdline, MAX_CMD_LEN, MAX_CMD_LEN, L"\"%ls\\ssh-shellhost.exe\" ---pty %ls", __wprogdir, cmd_w);
-	/* 
-	 * In PTY mode, ssh-shellhost takes stderr as control channel
-	 * TODO - fix this and pass control channel pipe as a command line parameter
-	 */
-	si.hStdError = ttyh;
-	debug3("pty commandline: %ls", pty_cmdline);
+	do {
+		if (is_conpty_supported()) {
+			ret = CreateConPty(cmd_w, (short)si.dwXCountChars, (short)si.dwYCountChars, si.hStdInput, si.hStdOutput, ttyh, &pi);
```

Comment:
In CreateConPty() handles are not closed.. We need to fix that

Reply:
I combined the conpty into exec_command_with_pty and close handles in exec_command_with_pty.